core/shared/src/main/scala/zio/config/ConfigDocsModule.scala

```diff
@@ -600,6 +607,14 @@ trait ConfigDocsModule {
             )
           )
 
+        case Config.Switch(c, map) =>
+          ConfigDocs.DynamicMap(
+            loop(descriptions, c, latestPath, alreadySeen),
+            map.map { case (k, v) =>
+              k.toString -> loop(descriptions, v, latestPath, alreadySeen)
```

Comment:
In most cases `k` will be `String`, but I wonder if there could be a better support for polymorphic key type of `Config.Switch[A, B]`.

Reply:
It's going to be hard at DynamicMap level in zio core, as this key gets propagated all over.